src/vm/prestub.cpp

```diff
@@ -731,31 +731,52 @@ Stub * CreateUnboxingILStubForSharedGenericValueTypeMethods(MethodDesc* pTargetM
 
     // 2. Emit the method body
     mdToken tokPinningHelper = pCode->GetToken(MscorlibBinder::GetField(FIELD__PINNING_HELPER__M_DATA));
-    
+
     // 2.1 Push the thisptr
     // We need to skip over the MethodTable*
-    // The trick below will do that. 
+    // The trick below will do that.
     pCode->EmitLoadThis();
     pCode->EmitLDFLDA(tokPinningHelper);
+    unsigned int numArgs = msig.NumFixedArgs() + 1; // +1 is for thisptr
 
-    // 2.2 Push the hidden context param 
-    // The context is going to be captured from the thisptr
-    pCode->EmitLoadThis();
-    pCode->EmitLDFLDA(tokPinningHelper);
-    pCode->EmitLDC(Object::GetOffsetOfFirstField());
-    pCode->EmitSUB();
-    pCode->EmitLDIND_I();
+#if defined(_TARGET_X86_)
+    if (numArgs < NUM_ARGUMENT_REGISTERS)
```

Comment:
I see that a similar fix was done in CreateInstantiatingILStub some time ago as well. It should be fixed there as well to not check NUM_ARGUMENT_REGISTERS - if you agree that it is the right fix.